src/lens.js

```diff
@@ -1,51 +1,31 @@
 var _curry2 = require('./internal/_curry2');
+var map = require('./map');
 
 
 /**
- * Creates a lens. Supply a function to `get` values from inside an object, and a `set`
- * function to change values on an object. (n.b.: This can, and should, be done without
- * mutating the original object!) The lens is a function wrapped around the input `get`
- * function, with the `set` function attached as a property on the wrapper. A `map`
- * function is also attached to the returned function that takes a function to operate
- * on the specified (`get`) property, which is then `set` before returning. The attached
- * `set` and `map` functions are curried.
+ * Returns a lens for the given getter and setter functions. The getter "gets"
+ * the value of the focus; the setter "sets" the value of the focus. The setter
+ * should not mutate the data structure.
  *
  * @func
  * @memberOf R
  * @category Object
- * @sig (k -> v) -> (v -> a -> *) -> (a -> b)
- * @param {Function} get A function that gets a value by property name
- * @param {Function} set A function that sets a value by property name
- * @return {Function} the returned function has `set` and `map` properties that are
- *         also curried functions.
+ * @typedef Lens s a = Functor f => (a -> f a) -> s -> f s
+ * @sig (s -> a) -> ((a, s) -> s) -> Lens s a
+ * @param {Function} getter
+ * @param {Function} setter
+ * @return {Lens}
+ * @see R.view, R.set, R.over, R.lensIndex, R.lensProp
```

Comment:
I'm taking full advantage of our newly acquired ability to reference multiple other functions. :)